src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -179,6 +181,32 @@ def sort_code_mapping_configs(
                     sorted_configs.insert(0, config)
         return sorted_configs
 
+    def get_commit_sha(
+        self,
+        integration: Integration,
+        organization_id: str,
+        line_no: int,
+        filepath: str,
+        repository: Repository,
+        branch: str,
+    ) -> Optional[str]:
+        commitSha = None
+        integrationInstallation = integration.get_installation(organization_id=organization_id)
+        try:
+            git_blame_list = integrationInstallation.get_blame_for_file(
+                repository,
+                filepath,
+                branch,
+                line_no,
+            )
+            for blame in git_blame_list:
+                if blame["startingLine"] <= line_no <= blame["endingLine"]:
+                    commitSha = str(blame["commit"]["oid"])
+                    break
+        except Exception:
```

Comment:
Could you please move the try/except block a level higher? (around the `result["commitSha"] = self.get_commit_sha()` call) I'm going to propose from here onward that the caller is the one that needs to handle errors. That way we do not handle side-effects within the function (e.g. calling logger.exception).